docs/events/glides/com.mbta.ctd.glides.vehicle_trip_assignments.v1.mdx

```diff
@@ -0,0 +1,176 @@
+---
+sidebar_label: glides.vehicle_trip_assignments
+sidebar_position: 2
+---
+
+import EventSchemaPath from "@site/src/components/EventSchemaPath.tsx";
+import Examples from "@site/src/components/Examples.tsx";
+import example from "@site/examples/glides-events/vehicle_trip_assignments.v1.json";
+
+# vehicle_trip_assignments
+
+event type: `com.mbta.ctd.glides.vehicle_trip_assignments.v1`
+
+Describes which vehicle is operating on which trip at the current moment. An event is sent whenever a vehicle is assigned or unassigned from a trip. One event may batch multiple assignments.
+
+Does not provide predictions for which vehicles will be assigned to future trips.
+
+See the the [Updating Assignments](#updating-assignments) section for how vehicles and trips are assigned and unassigned.
+
+## Data
+
+Fields in the event's `data`:
+
+- `assignments` (array of [VehicleTripAssignment](#vehicletripassignment)): The list of vehicles and trips with new assignments.
+
+Note there is no [Metadata](index.mdx#metadata) field like in other events, because these events are more likely to be caused by updates from the realtime data than by data inputted into the Glides app by Green Line staff.
+
+The order of `assignments` is not meaningful.
+
+### TripKey
+
+Fields:
+
+- `serviceDate` (RFC3999 date): the service date for the trip.
+- `tripId` (string): ID, unique within the service date. Trip IDs for scheduled trips are intended to match GTFS trip IDs, but may not.
+- `revenue` (`"revenue"` | `"nonrevenue"`): Whether the trip is a revenue trip.
+- `scheduled` (`"scheduled"` | `"added"`): Whether this is a trip that Glides has in its schedule or that Glides added.
+
+This is a different structure than the [`TripKey` used in `trips_updated`](com.mbta.ctd.glides.trips_updated.v1.mdx#tripkey). Added and scheduled trips use the same format, and are differentiated by the `scheduled` field instead of by using different formats.
+
+- If `scheduled` is `"scheduled"`, then this TripKey's `tripId` will equal the `trips_updated` TripKey's `tripId`.
+- If `scheduled` is `"added"`, then this TripKey's `tripId` will equal the `trips_updated` TripKey's `glidesId`.
+- (In both cases, the `serviceDate` must also match in order to know the two TripKeys refer to the same trip.)
+
+The difference with the TripKey in `trips_updated` is because `trips_updated` has to give enough information about a trip that a consumer can understand it even without recognizing the trip ID, while `vehicle_trip_assignments` assumes that you can look up information in `trips_updated` or in `GTFS`, so doesn't give as much information as possible about the trip. (Though consumers should keep in mind that order of events between two event streams is not guaranteed, and a trip that's referenced in this feed may not have appeared in `trips_updated`.)
```

Comment:
Am I following this correctly that we need to look at both `scheduled` and `trip_id` to know what field of `trips_updated` to match this to?

In the event of an added trip, what will the `trip_id` of `trips_updated` be? It would be nice if there was a single shared identifier.

Reply:
You need to look at `scheduled` to see which field to compare to, and `trip_id` to see which value to look for there.

In an added trip, `trips_updated` doesn't have a `trip_id` field, the `TripKey` is `{serviceDate, glides_id}`. ([docs](https://mbta.github.io/schemas/events/glides/com.mbta.ctd.glides.trips_updated.v1#added-trip))

It would be nice to have a single field to compare to. That's a flaw in the `trips_updated.v1` spec, because it was written before we started using GTFS ids. Now I'm dreaming of a `trips_updated.v2` that combines the two `TripKey` types into one and has a single `trip_id` field, but that would be a separate change, and this `vehicle_trip_assignments.v1` spec would be forward-compatible with that change.